### ISSUE TITLE
Add documentation/configuration option about email verification

### ISSUE THREAD

**Issue by marcelovilla** [MetroStar/nebari-self-registration#15]
Right now, [the README states that](https://github.com/MetroStar/nebari-self-registration/blob/aca7fd3d970caff325ffbb7d2774ac089360f73f/README.md?plain=1#L57):

> After you have entered a new password, you will receive a verification email.

However, for a fresh deployment of Nebari with the plugin, that's not the case. I'm assuming there's some kind of external configuration that has to be done (e.g., configuring a SMTP server in Keycloak) before the verification email gets sent.

Besides documenting this, would it make sense to expose it as a configuration option and potentially have the plugin configure the SMTP server automatically given some parameters?

**Comment by kenafoster** [comment]
Good catch.  For a first step, I will update the README so that it says the user will only receive a verification email if the email validation is currently enabled (with a link to the page I'm also working on about enabling outgoing SMTP)

Since email validation and SMTP configuration are core configuration items, I think it would make more sense for it to be configurable in the base Nebari config than through an extension.

**Comment by viniciusdc** [comment]
This might be related @marcelovilla https://github.com/nebari-dev/nebari-docs/pull/463

> Since email validation and SMTP configuration are core configuration items, I think it would make more sense for it to be configurable in the base Nebari config than through an extension.

This is reasonable. We might have already exposed a custom override capability to Keycloak's settings, but that is not well documented as we probably never used it. 

Also, a bit out of the scope of this discussion, but regarding exposing some configuration to the user, I like the example HTML template Oauth0 provides. Maybe it would be a nice addition to enable styling or provide your own:

![image](https://github.com/MetroStar/nebari-self-registration/assets/51954708/4b537689-e4dc-4476-b817-0b08320a552b)
 and the respective rendered email:
![image](https://github.com/MetroStar/nebari-self-registration/assets/51954708/e4110eca-2a08-4ff2-b17f-819aac5824de)

**Comment by viniciusdc** [comment]
Also, a bit out of this discussion scope, but regarding exposing some configuration to the user here,  maybe it would be a nice addition to enable styling or provide your own custom email verification/token HTML:

![image](https://github.com/MetroStar/nebari-self-registration/assets/51954708/e4110eca-2a08-4ff2-b17f-819aac5824de)

**Comment by kenafoster** [comment]
Did a little digging here - it's possible to develop a custom set of email templates in Freemarker and then mount them at /opt/jboss/keycloak/themes/nebari .  We could also use the `theme.jupyterhub` portion of the config file to generate a template set with branding that matches the user's custom JHub theme.
